src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -0,0 +1,53 @@
+package com.kodilla.ecommercee.controller;
+
+import com.kodilla.ecommercee.domain.dto.CartDto;
+import com.kodilla.ecommercee.domain.dto.CartItemDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@CrossOrigin("*")
+@RequestMapping(value = "/carts")
+public class CartController {
+    
+    @PostMapping
+    public ResponseEntity<Void> createCart() {
+        List<CartItemDto> cart = new ArrayList<>();
+        //post to repository//
+        return ResponseEntity.ok().build();
+    }
+    
+    @GetMapping("/{cartId}")
+    public ResponseEntity<CartDto> getCart(Long cartId) {
+        //get from repository//
+        //map to Dto//
+        CartDto currentCard = new CartDto(cartId, new ArrayList<>());
+        return ResponseEntity.ok(currentCard);
+    }
+
+    @PostMapping(value = "/addItem")
+    public ResponseEntity<Void> addCartItem(@RequestBody CartItemDto cartItemDto) {
+        //post item into Cart//
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping(value = "/{cartItemId}")
+    public ResponseEntity<Void> removeCartItem(@PathVariable Long cartItemId) {
+        return ResponseEntity.ok().build();
+    }
```

Comment:
@DeleteMapping(value ="/deleteItem/{cartItemId}")
 public ResponseEntity<Void> removeCartItem(@PathVariable Long cartItemId, @RequestParam Long cartId) {
 return ResponseEntity.ok().build();
    }

Reply:
ok

Reply:
I though it is for deleting cart from table. If so, for that, as far as I know, you need only cartId (I assumed cartItemId is that).
For deleting product from the list we do not delete record from table, therefore we are not using "Delete" but "Put" which is for update. We update list of product for Cart by removing item from the list.

Reply:
maybe Put or Patch but I would leave Delete, because after all, it is an operation to delete something( one item from list), at most it will change later